modules/kotest/src/main/kotlin/org/testcontainers/containers/ContainerPerSpecListener.kt

```diff
@@ -0,0 +1,49 @@
+package org.testcontainers.containers
+
+import io.kotest.core.listeners.TestListener
+import io.kotest.core.spec.Spec
+import io.kotest.core.test.TestCase
+import io.kotest.core.test.TestResult
+import kotlin.reflect.KClass
+
+
+/**
+ * [ContainerPerSpecListener] starts the given [genericContainers] before execution of any test in the spec
+ * and stops after execution of all tests.
+ *
+ * [genericContainers] can any of [GenericContainer] [ContainerisedDockerCompose] [LocalStackContainer] etc.
+ *
+ * This should be use when you want to a single container for all test in a single test class.
+ *
+ * @see
+ * [ContainerPerSpecListener]
+ *
+ * Usage:
```

Comment:
We should have a docs section with code-include for this. Code could be in an examples module.

Reply:
OK I will do that

Reply:
@kiview i had move kotest example usage to docs, but currently gradle is unable to resolve ContainerPerTestListener and ContainerPerSpecListener import statement in test, could you please have a look at the my build.gradle file and let me where exactly i meshed up

Reply:
@kiview The other thing in my mind is that if you look at my current test in docs folder they are very concrete tests with respect to my earlier(mocking based) test in kotest folder under modules, how about replacing the mocking based test with these concrete test and provided a simple readme if require for documentation.